packages/adapter/src/lib/adapter/translationManager.ts

```diff
@@ -0,0 +1,99 @@
+import type { Client as ObjectsClient } from '@iobroker/db-objects-redis';
+import path from 'node:path';
+import fs from 'node:fs/promises';
+
+interface TranslationManagerOptions {
+    /** The objects DB instance */
+    objects: ObjectsClient;
+    /** The config i18n directories, as absolute paths */
+    configDirs: string[];
+}
+
+/** The default language to be used */
+const DEFAULT_LANG: ioBroker.Languages = 'en';
+
+export class TranslationManager {
+    /** The objects DB instance */
+    private readonly objects: ObjectsClient;
+    /** The config i18n directories as absolute paths */
+    private readonly configDirs: string[];
+    /** The actual translations */
+    private translations = new Map<string, string>();
+    /** The currently loaded language */
+    private loadedLanguage?: ioBroker.Languages;
+
+    constructor(options: TranslationManagerOptions) {
+        this.objects = options.objects;
+        this.configDirs = options.configDirs;
+    }
+
+    /**
+     * Initialize the Translator instance
+     * This determines the locale and loads the translations from the config file
+     */
+    public async init(): Promise<void> {
+        const sysConfObj = await this.objects.getObject('system.config');
+
+        await this.updateLanguage(sysConfObj?.common.language || DEFAULT_LANG);
+    }
+
+    /**
+     * Set the language and load the translations
+     *
+     * @param lang the language to load the translations for
+     */
+    public async updateLanguage(lang: ioBroker.Languages): Promise<void> {
+        if (lang === this.loadedLanguage) {
+            return;
+        }
+
+        this.translations.clear();
+
+        this.loadedLanguage = lang;
+        let allTranslations: Record<string, string> = {};
+
+        for (const configDir of this.configDirs) {
+            let content: string;
+
+            try {
+                content = await fs.readFile(path.join(configDir, `${lang}.json`), { encoding: 'utf8' });
```

Comment:
do we not have fsextra here, so we can use better methods?

Reply:
We have, however as now (I think) all of the methods we use have native promise based methods I thought of preferring these in the long run.